### PR TITLE
Introduce server application

### PR DESCRIPTION
**Pull request by ximura**
Create separated configuration file for dedicated server (on Linux and Windows)
Client and dedicated server can have different list of Game Modules
Client and dedicated server should have same gems list

Configuration file named server.xml it stored in <game_folder>/config (same folder as game.xml)
Known Issue: lmbr.exe projects populate-appdescriptors dosn't change server.xml (it should be filled manually)

### CODE REVIEW

File: dev/Code/Framework/AzGameFramework/AzGameFramework/Application/ServerApplication.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+*
+*/
+
+#include "ServerApplication.h"
+
+namespace AzGameFramework
+{
+    ServerApplication::ServerApplication()
+    {
+    }
```
Comment (SSPkrolik) [inline dev/Code/Framework/AzGameFramework/AzGameFramework/Application/ServerApplication.cpp]: I think we can eliminate empty constructor or at least explicitly mark it as default.

File: dev/Code/Framework/AzGameFramework/AzGameFramework/Application/ServerApplication.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    ServerApplication::~ServerApplication()
+    {
+    }
```
Comment (SSPkrolik) [inline dev/Code/Framework/AzGameFramework/AzGameFramework/Application/ServerApplication.cpp]: The same with empty destructor

File: dev/Code/Framework/AzGameFramework/AzGameFramework/Application/ServerApplication.h [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+{
+    class ServerApplication
+        : public GameApplication
+    {
+    public:
+
+        AZ_CLASS_ALLOCATOR(ServerApplication, AZ::SystemAllocator, 0);
+
+        ServerApplication();
+        ~ServerApplication();
```
Comment (SSPkrolik) [inline dev/Code/Framework/AzGameFramework/AzGameFramework/Application/ServerApplication.h]: `ServerApplication() = default;`

File: dev/Code/Framework/AzGameFramework/AzGameFramework/Application/ServerApplication.h [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+*/
+
+#ifndef AZGAMEFRAMEWORK_SERVERAPPLICATION_H
+#define AZGAMEFRAMEWORK_SERVERAPPLICATION_H
+
+#include <AzCore/base.h>
+#include "GameApplication.h"
+
+#pragma once
+
```
Comment (rustamserg) [inline dev/Code/Framework/AzGameFramework/AzGameFramework/Application/ServerApplication.h]: We can replace old c style include guard with pragma once and put it before includes

**Pull request closed without merging**